test/compress/destructuring.js

```diff
@@ -639,3 +639,25 @@ issue_2044_ecma_6_beautify: {
     }
     expect_exact: "({x: a = 1, y = 2 + b, z = 3 - c} = obj);"
 }
+
+issue_2140: {
+    options = {
+        unused: true,
+    }
+    input: {
+        !function() {
+            var s = {}, t = {};
+            [t.a] = [42];
+            console.log(t.a);
```

Comment:
This is basically https://github.com/mishoo/UglifyJS2/issues/2140#issuecomment-310296079 - I thought doing away with `toplevel` and being more specific helps.

But I've probably screwed up `options` - I meant to check that against `harmony`, but I got distracted :sweat_smile: 

Thanks for catching this!